Pre-order.c

```diff
@@ -0,0 +1,49 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+
+struct node{
+    struct  node *lchild;
+    int data;
+    struct node *rchild;
+}*root = NULL;
+
+struct node *insert(struct node *passnode, int key){
+    struct node *t = NULL;
+    if(passnode==NULL){
+        temp = (struct node *)malloc(sizeof(struct node));
+        temp->data = key;
+        temp->lchild = temp->rchild = NULL;
+
+        return temp;
+    }
+    if(key<passnode->data){
+        passnode->lchild = insert(passnode->lchild, key);
+    }
+    else{
+        passnode->rchild = insert(passnode->rchild, key);
+    }
+
+    return passnode;
+}
+
+void preorder(struct node* passnode)
+{
+    if (p == NULL)
```

Comment:
give proper name  (p will be show as undeclared)

Reply:
Ok, its passnode